vector/src/main/java/im/vector/app/features/autocomplete/member/AutocompleteMemberPresenter.kt

```diff
@@ -71,6 +71,23 @@ class AutocompleteMemberPresenter @AssistedInject constructor(context: Context,
         val members = room.getRoomMembers(queryParams)
                 .asSequence()
                 .sortedBy { it.displayName }
+                .disambiguate()
         controller.setData(members.toList())
     }
 }
+
+private fun Sequence<RoomMemberSummary>.disambiguate(): Sequence<RoomMemberSummary> {
+    val displayNames = hashMapOf<String, Int>().also { map ->
+        for (item in this) {
+            item.displayName?.lowercase()?.also { displayName ->
+                map[displayName] = map.getOrPut(displayName, { 0 }) + 1
+            }
+        }
+    }
```

Comment:
Why not simply write:
`val displayNames = groupBy { it.displayName?.lowercase() }.mapValues { it.value.size }`

Also, I do not think we disambigate display name regarding the case, for instance in the timeline, for sender name. But why not, I think it's worth it here.

Reply:
I didn't use groupBy and mapValues because they each loop over the sequence, and because groupBy creates a List for every displayName. I didn't want there to be a big performance impact for disambiguation (especially since it loops over thousands of people in big rooms), therefore I wrote it differently so it would loop only once without creating a list for each displayName.

Reply:
I see, thanks for the feedback